src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpResponseParser.cs

```diff
@@ -0,0 +1,245 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Runtime.InteropServices;
+using System.Text;
+
+using SafeWinHttpHandle = Interop.WinHttp.SafeWinHttpHandle;
+
+namespace System.Net.Http
+{
+    internal static class WinHttpResponseParser
+    {
+        private const string EncodingNameDeflate = "DEFLATE";
+        private const string EncodingNameGzip = "GZIP";
+        private const string HeaderNameContentEncoding = "Content-Encoding";
+        private const string HeaderNameContentLength = "Content-Length";
+        private const string HeaderNameSetCookie = "Set-Cookie";
+        private static readonly string[] s_HttpHeadersSeparator = { "\r\n" };
+
+        public static HttpResponseMessage CreateResponseMessage(
+            WinHttpRequestState state,
+            bool doManualDecompressionCheck)
+        {
+            HttpRequestMessage request = state.RequestMessage;
+            SafeWinHttpHandle requestHandle = state.RequestHandle;
+            CookieUsePolicy cookieUsePolicy = state.Handler.CookieUsePolicy;
+            CookieContainer cookieContainer = state.Handler.CookieContainer;
+            var response = new HttpResponseMessage();
+            bool useDeflateDecompression = false;
+            bool useGzipDecompression = false;
+
+            // Get HTTP version, status code, reason phrase from the response headers.
+            string version = GetResponseHeaderStringInfo(requestHandle, Interop.WinHttp.WINHTTP_QUERY_VERSION);
+            if (string.Compare("HTTP/1.1", version, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                response.Version = new Version(1, 1);
+            }
+            else if (string.Compare("HTTP/1.0", version, StringComparison.OrdinalIgnoreCase) == 0)
+            {
+                response.Version = new Version(1, 0);
```

Comment:
FYI, I did that as part of #3199 (among other changes), but I'll have to fix up any merge conflicts in that PR once this is merged.